src/bin/pgcopydb/ld_stream.c

```diff
@@ -896,6 +849,15 @@ stream_write_json(LogicalStreamContext *context, bool previous)
 		}
 	}
 
+	if (metadata->action == STREAM_ACTION_BEGIN)
+	{
+		privateContext->transactionInProgress = true;
+	}
+	else if (metadata->action == STREAM_ACTION_COMMIT)
+	{
+		privateContext->transactionInProgress = false;
+	}
+
```

Comment:
Do we need to add a case for STREAM_ACTION_ROLLBACK? if not, we need to add a comment about why...